Mathlib/Analysis/Normed/Operator/BoundedLinearMaps.lean

```diff
@@ -217,6 +260,58 @@ theorem isBoundedLinearMap_continuousMultilinearMap_comp_linear (g : G →L[𝕜
 
 end
 
+section
+
+variable (𝕜) in
+abbrev ContinuousLinearMap.mk' (f : E → F) (hf : IsBoundedLinearMap 𝕜 f) : E →L[𝕜] F :=
+  hf.toContinuousLinearMap
+
+open Lean.Parser.Term in
+macro:max "fun " x:funBinder " ↦L[" R:term "] " b:term : term =>
```

Comment:
I don't think `IsBoundedLinearMap` is the right thing to use here. If we want to claim this notation, it should work for general linear maps between topological vector spaces, which means you only need `IsLinearMap f` + `Continuous f`, not the stronger `IsBoundedLinearMap`.

Reply:
Options I see:
- synthesize both proofs but `IsLinearMap` seems effectively deprecated. Also this will be roughly twice as slow than other options.
- introduce new `IsContinuousLinearMap` this will require developing whole slew of new API (this is my approach in SciLean)
- stick with `IsBoundedLinearMap` for now. Will this case unpleasant refactor when we decide to generalize in the future?

Reply:
So are you suggesting option one?

Reply:
I think option 2 starts to approximate

> Obviously the "cleaner" thing to do would be to redefine `LinearMap`, but that's a massive refactor we should not attempt unless we want to commit to doing it globally.

which perhaps warrants a Zulip discussion.


Option 1 seems like the best approach to me. Can you tell fun_prop to prioritize `IsBoundedLinearMap` -> `IsLinearMap` to speed up search in the common case, or will it then search for `IsBoundedLinearMap f` twice?

Reply:
Ahh fair point. I mostly care about calculus which still mostly work only over normed spaces so it didn't seemed to me as a problem

Reply:
The approach we can take is for `ContinuousLinearMap.mk'` to take `IsLinearMap 𝕜 f  ∧ Continuous f` and write a specialized tactic that calls `fun_prop` to infer `IsBoundedLinearMap` and if that fails it tries to look for linearity and continuity. However, this will mean triple the cost if the search for `IsBoundedLinearMap` fails.

Reply:
The notation is now based on `ContinuousLinearMap.mk'` that takes proof of `IsLinearMap R f ∧ Continuous f` and there is a new tactic `is_clm` that tries to prove `IsBoundedLinearMap` first with `fun_prop` and if it fails it tries to prove linearity and continuity separately.